feature-libs/checkout/components/components/shipping-address/shipping-address.component.ts

```diff
@@ -109,19 +109,27 @@ export class ShippingAddressComponent implements OnInit, OnDestroy {
   }
 
   getSupportedAddresses(): Observable<Address[]> {
-    if (this.isAccountPayment) {
+    if (
+      this.isAccountPayment &&
+      this.checkoutCostCenterService &&
+      this.userCostCenterService
+    ) {
       return this.checkoutCostCenterService.getCostCenter().pipe(
         distinctUntilChanged(),
         switchMap((selected) => {
           this.doneAutoSelect = false;
-          return this.userCostCenterService.getCostCenterAddresses(selected);
+          return (
+            this.userCostCenterService?.getCostCenterAddresses(
```

Comment:
I think we should not need the "?" in "userCostCenterService?." if it's value iis validatted in the "if" att liine 112.

If a strict mode error happens without the "?." perhaps try to test `this.userCostCenterService !== undefined` in the "if".

Reply:
without this `?`, it has error, even we have this check `this.userCostCenterService !== undefined`. Unless we check this `this.userCostCenterService !== undefined` inside `switchMap`, then it is same thing as `this.userCostCenterService?.getCostCenterAddresses`.

Reply:
Alright.